### PR TITLE
[Feature] : Update Health Metrics Event Name

### PR DESCRIPTION
**Pull request by radityagumay**
## Abstract

Health Metrics is the way we organized the performance of the Clickstream where we push events to the backend. Health Metrics has been serving us @ Gojek for the last 1,5 years and it performs very well. However, as we open source the clickstream, we also want the communities to get the power of the health metrics; this is needed as the first version of Health Metrics is coupled with the Gojek environment.

Please see the detail of RFC over here https://github.com/gojek/clickstream-android/discussions/18

### CODE REVIEW

File: clickstream-health-metrics/src/main/kotlin/clickstream/health/internal/DefaultCSHealthEventProcessor.kt [hunk 1]
```diff
@@ -219,119 +180,41 @@ public class DefaultCSHealthEventProcessor(
...
 
-    private suspend fun logAndDeleteEvents(events: List<CSHealthEvent>) {
-        logger.debug { "CSHealthEventProcessor#pushEvents" }
-
+    private suspend fun trySendAndDeleteEvents(events: List<CSHealthEvent>) {
         events.forEach {
-            if (!healthEventConfig.isTrackedViaClickstream(it.eventName)) {
+            // send to upstream
+            if (healthEventConfig.isTrackedForExternal() || healthEventConfig.isTrackedForBoth()) {
                 healthEventLogger.logEvent(eventName = it.eventName, eventData = it.eventData())
```
Comment (radityagumay) [inline clickstream-health-metrics/src/main/kotlin/clickstream/health/internal/DefaultCSHealthEventProcessor.kt:219]: need testing this part

Comment (radityagumay) [reply clickstream-health-metrics/src/main/kotlin/clickstream/health/internal/DefaultCSHealthEventProcessor.kt]: done

File: clickstream-health-metrics-api/build.gradle.kts [hunk 2]
```diff
@@ -35,4 +35,6 @@ android {
 dependencies {
     // Clickstream
     compileOnly(files("$rootDir/libs/proto-sdk-1.18.6.jar"))
+
+    implementation("androidx.annotation:annotation:1.1.0")
```
Comment (kshitij6325) [inline clickstream-health-metrics-api/build.gradle.kts:39]: Can we add this to buildSrc and use it from there?

File: clickstream-health-metrics-api/src/main/kotlin/clickstream/health/constant/CSEventNamesConstant.kt [hunk 3]
```diff
@@ -1,23 +1,136 @@
...
+public enum class CSEventNamesConstant(public val value: String) {;
+
+    public enum class Instant(public val value: String) {
+        /**
+         * Tracks the instances where the connection gets dropped.
+         *
+         * Type: Instant
+         * Priority: Critical
+         */
+        ClickStreamConnectionDropped("Clickstream Connection Dropped"),
```
Comment (kshitij6325) [inline clickstream-health-metrics-api/src/main/kotlin/clickstream/health/constant/CSEventNamesConstant.kt:15]: Can we drop the 'ClickStream' prefix? It is should be clear from parent class.

**Pull request merged**